### PR TITLE
feat: add mocking/stubbing sample

### PR DESCRIPTION
**Pull request by G0maa**
- Auto generating mocks using `node-fetch` as an example
- Fixes https://github.com/keploy/keploy/issues/428

### CODE REVIEW

**Comment by Nishant2907** [comment]
@re-Tick Can you please review this? 
@G0maa Please wait while the maintainers review this. 
Once the review is done, kindly add this to the completed task list in the Gsoc repo!

File: mocking-stubbing/test/test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+describe('Generating of mocks for external dependencies', function () {
+  before(() => {
+    // Setting mode to record before using fetch
+    NewContext({ Mode: 'record', Name: 'testing-auto-mocks' });
+  });
+
+  test('Generating node-fetch mocks', () => {
+    fetch('https://reqres.in/api/users/2')
+      .then((res) => res.text())
+      .then((text) => rs.send(text));
```
Comment (re-Tick) [inline mocking-stubbing/test/test.js:15]: Please add assertions to check for the expected response.

Comment (G0maa) [reply mocking-stubbing/test/test.js]: - @re-Tick, done, could you check the new commit?
- off-topic: Currently `NewContext` returns undefined, which makes me unable to check if `Keploy` worked well or not, it'd be nice to have that ability.

Comment (re-Tick) [reply mocking-stubbing/test/test.js]: LGTM.

Comment (re-Tick) [reply mocking-stubbing/test/test.js]: > * @re-Tick, done, could you check the new commit?
> * off-topic: Currently `NewContext` returns undefined, which makes me unable to check if `Keploy` worked well or not, it'd be nice to have that ability.

Yes it makes sense to return an error for failure. If you like to work on this, you can make an issue and propose your changes for discussion.

**Comment by re-Tick** [comment]
Hey @G0maa, Please add PR link in the corresponding issue of TS-SDK task list (GSOC/2023).

**Comment by G0maa** [comment]
Hello @re-Tick, sorry for being late... totally forgot about adding PR link, here: https://github.com/keploy/gsoc/pull/40

**Comment by re-Tick** [comment]
This PR have been reviewed and accepted but not merged until GSOC. Since, the issues are open for all and changes will be considered for selection , so anyone can propose their changes. If the changes are same then, PR will considered on first come basis

**Pull request closed without merging**